app/models/container_image.rb

```diff
@@ -92,9 +92,22 @@ def annotate_deny_execution(causing_policy)
     ext_management_system.annotate(
       "image",
       digest,
+      annotations
+    )
+  end
+
+  def annotate_allow_execution(causing_policy)
+    annotate_image({
+      "security.manageiq.org/successful-policy" => causing_policy,
+      "images.openshift.io/allow-execution"     => "true"
```

Comment:
I see what you're asking now. It seems you don't want to overwrite an annotation. Once it's non-compliant it should stay that way. I don't see how adding another annotation helps. Now you could have both allow and deny annotations co-existing. A single boolean is cleanest implementation. How about a single method that accepts a pass/fail boolean?

```
  def annotate_execution(causing_policy, deny_execution)
    annotate_image({
      "security.manageiq.org/policy"        => causing_policy,
      "images.openshift.io/deny-execution"  => deny_execution
    })
```